browser/permissions/google_sign_in_permission_browsertest.cc

```diff
@@ -0,0 +1,556 @@
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  void TearDownOnMainThread() override { prompt_factory_.reset(); }
+
+  permissions::MockPermissionPromptFactory* prompt_factory() {
+    return prompt_factory_.get();
+  }
+
+  void DefaultBlockAllCookies() {
+    brave_shields::SetCookieControlType(
+        content_settings(), browser()->profile()->GetPrefs(),
+        brave_shields::ControlType::BLOCK, GURL());
+  }
+
+  void DefaultAllowAllCookies() {
+    brave_shields::SetCookieControlType(
+        content_settings(), browser()->profile()->GetPrefs(),
+        brave_shields::ControlType::ALLOW, GURL());
+  }
+
+  void AllowCookies(const GURL& url) {
+    brave_shields::SetCookieControlType(content_settings(),
+                                        browser()->profile()->GetPrefs(),
+                                        brave_shields::ControlType::ALLOW, url);
+  }
+
+  void BlockThirdPartyCookies(const GURL& url) {
+    brave_shields::SetCookieControlType(
+        content_settings(), browser()->profile()->GetPrefs(),
+        brave_shields::ControlType::BLOCK_THIRD_PARTY, url);
+  }
+
+  void BlockCookies(const GURL& url) {
+    brave_shields::SetCookieControlType(content_settings(),
+                                        browser()->profile()->GetPrefs(),
+                                        brave_shields::ControlType::BLOCK, url);
+  }
+
+  void NavigateToPageWithFrame(const GURL& url) {
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  }
+
+  void ExpectCookiesOnHost(const GURL& url, const std::string& expected) {
+    EXPECT_EQ(expected, content::GetCookies(browser()->profile(), url));
+  }
+
+  void NavigateFrameTo(const GURL& url, const std::string& id = "test") {
+    content::WebContents* web_contents =
+        browser()->tab_strip_model()->GetActiveWebContents();
+    EXPECT_TRUE(NavigateIframeToURL(web_contents, id, url));
+  }
+
+  content::WebContents* contents() {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  void SetGoogleSignInPref(bool value) {
+    browser()->profile()->GetPrefs()->SetBoolean(kGoogleLoginControlType,
+                                                 value);
+  }
+
+  void ClickButtonWithId(const std::string& id) {
+    std::string click_script = base::StringPrintf(
+        R"(
+        new Promise(async (resolve, reject) => {
+            try {
+              const button = document.getElementById('%s');
```

Comment:
can be simplified with `content::JsReplace`. It will auto wrap with `''` and JS-encode the string value.